addons/mail/static/src/discuss/gif_picker/composer_patch.js

```diff
@@ -0,0 +1,49 @@
+/* @odoo-module */
+
+import { patch } from "@web/core/utils/patch";
+import { Composer } from "@mail/composer/composer";
+import { useRef, useState } from "@odoo/owl";
+import { useGifPicker } from "./gif_picker_hook";
+import { GifPicker } from "./gif_picker";
+import { useService } from "@web/core/utils/hooks";
+
+Composer.components.GifPicker = GifPicker;
+/** @type {Composer} */
+const composerPatch = {
+    setup() {
+        this._super();
+        this.gifRef = useRef("gif");
+        this.gifPickerState = useState({
+            gifKeyboardOpen: false,
+        });
+        this.ui = useState(useService("ui"));
+        if (this.store.hasGifPickerFeature && !this.ui.isSmall) {
+            this.gifPicker = useGifPicker(this.gifRef, {
+                onSelected: this.sendGifMessage.bind(this),
+            });
+        }
+    },
+    onClickAddGif() {
+        this.gifPicker?.toggle();
+        if (!this.gifPicker && this.ui.isSmall) {
+            if (!this.gifPickerState.gifKeyboardOpen) {
+                this.gifPickerState.gifKeyboardOpen = true;
+            } else {
+                this.closeGifKeyboard();
+            }
+        }
+    },
+    closeGifKeyboard() {
+        this.gifPickerState.gifKeyboardOpen = false;
+    },
```

Comment:
This method is not worth it. I would just change the state explicitly where needed

Reply:
This method is also passed as a props callback to the gif picker.
This is because we need to close the gif keyboard when the user post a gif.